src/app/members/components/MemberHeader.tsx

```diff
@@ -9,17 +9,34 @@ import { usePathname } from "next/navigation";
 import { MemberBreadCrumbs } from "./MemberBreadCrumbs";
 import { Plus } from "lucide-react";
 import { Routes } from "@/lib/routes";
+import MemberContext, {
+  MemberContextType,
+} from "@/context/SearchContextProvider";
+import useMediaQuery from "@/hooks/use-media-query";
+import useDebounce from "@/hooks/useDebounce";
 
-export const MemeberHeader = () => {
-  const pathname = usePathname();
+export const MemberHeader = () => {
 
+  const pathname = usePathname();
+  const isSmallScreen = useMediaQuery("sm");
+  const [search, setSearch] = useState("");
+  const debounceValue = useDebounce(search, 500);
+  const { setSearchValue } = useContext(MemberContext) as MemberContextType;
   const [isSidebarOpen, setSidebarOpen] = useState<boolean>(false);
   const [isTitleHide, setIsTitleHide] = useState<boolean>(false);
 
   const toggleSidebar = () => {
     setSidebarOpen(!isSidebarOpen);
   };
 
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
Function name is way too generic. Should be something like `handleSearchQueryChange`